src/app/public/src/modules/link/link.directive.ts

```diff
@@ -1,31 +1,73 @@
-import { Directive, ElementRef, HostListener, Input, Renderer2 } from '@angular/core';
+import { Directive, ElementRef, HostListener, HostBinding, Input, Renderer2, OnChanges, AfterViewInit } from '@angular/core';
+
+import { LocationStrategy } from '@angular/common';
 
 import { StacheNavService } from '../nav';
 
+import { StacheRouteService, StacheWindowRef } from '../shared';
+
 @Directive({
   selector: '[stacheRouterLink]'
 })
-export class StacheRouterLinkDirective {
-  @Input()
-  public stacheRouterLink: string;
+export class StacheRouterLinkDirective implements OnChanges, AfterViewInit {
+
+  public _stacheRouterLink: string = '';
+
+  @Input('stacheRouterLink')
+  set stacheRouterLink(routerLink: string) {
+    if (routerLink === '.') {
+      this._stacheRouterLink = this.routerService.getActiveUrl();
+    } else {
+      this._stacheRouterLink = routerLink;
+    }
+  }
+
+  get stacheRouterLink() {
+    return this._stacheRouterLink;
+  }
 
   @Input()
   public fragment: string;
 
+  // the url displayed on the anchor element.
+  @HostBinding()
+  public href: string;
+
   constructor(
     private navService: StacheNavService,
+    private routerService: StacheRouteService,
     private el: ElementRef,
-    private renderer: Renderer2
+    private locationStrategy: LocationStrategy,
+    private renderer: Renderer2,
+    private windowRef: StacheWindowRef
   ) {
     this.renderer.setStyle(this.el.nativeElement, 'cursor', 'pointer');
   }
 
+  public ngOnChanges(changes: {}): any { this.updateTargetUrlAndHref(); }
+
+  public ngAfterViewInit() {
+    this.updateTargetUrlAndHref();
+  }
+
   @HostListener('click', ['$event'])
   public navigate(event: MouseEvent): void {
     event.preventDefault();
-    this.navService.navigate({
-      path: this.stacheRouterLink,
-      fragment: this.fragment
-    });
+    if (event.ctrlKey || event.metaKey || event.shiftKey) {
+      this.windowRef.nativeWindow.open(this.href);
```

Comment:
May want to avoid `open`, see: https://developer.mozilla.org/en-US/docs/Web/API/Window/open#Usability_issues

Reply:
Another thought, the nav service is already checking to see if the `path` is "global"; maybe you could tie into that somehow? https://github.com/blackbaud/stache2/blob/master/src/app/public/src/modules/nav/nav.service.ts#L17

(I created a `stacheRouterLink` on the demo page that pointed to an external URL and it seemed to work.)

Reply:
After doing a little bit of research it seems like window.open is really the only way to open a new tab/window in the browser. A lot of the warnings given in the mozilla link are in reference to issues that have since been resolved by native browser functionality, for instance: chrome and IE will automatically open a new tab instead of a browser window when ctrl+clicking. I believe that can be changed in the settings if the user would rather open a window. 

As far as I can tell there are no other alternatives that would provide us with the same level of functionality as the routerLink module. Let me know if you have more insight!

Reply:
```
  @HostListener('click', ['$event'])
  public navigate(event: MouseEvent): boolean {
    if (event.ctrlKey || event.metaKey || event.shiftKey) {
      return true;
    }

    this.navService.navigate({
      path: this.stacheRouterLink,
      fragment: this.fragment
    });
    return true;
  }
```
If we don't `preventDefault()` and we return true, it seems to allow the browser default actions for the anchor link.  This might let us avoid using `window.open`.  Is there any issue with doing this?  I think that's how Angular is handling it.

https://github.com/angular/angular/blob/5.2.10/packages/router/src/directives/router_link.ts#L217
https://github.com/angular/angular/blob/5.2.10/packages/router/src/directives/router_link.ts#L144

Reply:
@Blackbaud-BrandonJones 's approach seems to work. The preventDefault still needs to go outside of the if statement though. It's what is stopping the page from reloading every time you navigate to an in-page link.

Reply:
I can't seem to reproduce this?  With the return false at the end it won't reload.  Also which link are you clicking on to see this action happen?  TOC or links using the `nav.component` would need to have the `click` event removed from the component, otherwise it causes the reload?